### DISCUSSION TITLE
How-To: Enable Intel Fortran Debugging Support in VS Code

### DISCUSSION THREAD

**Discussion by gnikit** [krvajal/vscode-fortran-support#926] [Show and tell]
It is actually possible to use the debug adaptors that Intel ships for Visual Studio in Visual Studio Code as well, but it requires a bit of manual setup. I have written down the procedure to set it up here: https://gist.github.com/albertziegenhagel/6431811950864bd0009b6a1fa78e7f2b.

I would be more than happy if anyone tries that and let's me know if anything is unclear or whether there are any other issues.

---

_Originally posted by @albertziegenhagel as a comment in https://github.com/fortran-lang/vscode-fortran-support/discussions/915#discussioncomment-6239429_

**Comment by gnikit** [comment]
@albertziegenhagel This is quite promising, thanks for the how-to. If you see a way this could somehow be shipped with the VS Code extension we would be glad to include it. 

A somewhat relevant question is whether you know if this issue is also present with the C/C++ Intel compilers on Windows.
Are C++ developers using Intel compiler + VS Code also affected by this? If so, it would be a compelling argument with the C/C++ VS Code extension devs to natively support this.

**Reply by albertziegenhagel** [reply]
Although I am not sure, I would guess that some licensing terms would stop us from redistributing the binaries from the Intel packages. Maybe the best we could do is detect an available installation on a machine and do the steps from the how-to automatically. I could imagine something like

> "We have detected that the Intel Fortran integration has been installed for Visual Studio on this computer. Do you want to import that integration and make it available in Visual Studio Code as well?"

poping up as notification and the user can select yes/no/don't bother me again?

To the best of my knowledge there are no such problems with the C/C++ compilers. If I understand it correctly, the Intel C/C++ compilers are fully ABI compatible with MSVC, and hence the Visual Studio Debugger supports everything directly.

An other option would be write an own implementation of an expression evaluator extension for the VS debug engine. We might need to do something like that to support future LLVM like compilers like flang and/or LFortran anyways?! I guess that will depend on how they decide to specify their ABI and how they handle debug symbols. Maybe it will be possible to use a MiEngine based debugger like `lldb` via `cppdbg` for those compilers. To be honest I am not entirely familiar with the details here. It might be worth to take a look on how/whether debugging binaries compiled by the classic-flang compiler is supported on Windows.
